kotlin/workflow-ui-core/src/main/java/com/squareup/workflow/ui/Compatible.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2019 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.workflow.ui
+
+/**
+ * Returns true if [me] and [you] are instances of the same class, unless
+ * that class implements [Compatible]. In that case
```

Comment:
I don't thinks this is true? If `me` and `you` have different concrete types, the first case will fail and immediately return false.

Reply:
"that class" is the class that they are both instances of. I'll reword anyway.

Reply:
```
/**
 * Normally returns true if [me] and [you] are instances of the same class.
 * If that common class implements [Compatible], 
 * `me.`[canUpdateFrom][Compatible.isCompatibleWith]`(you)` must also be true.
 *
 * A convenient way to take control over the matching behavior of objects that
 * don't implement [Compatible] is to wrap them with [Named].
 */
```